crates/model/src/order.rs

```diff
@@ -357,6 +357,7 @@ pub struct OrderCreation {
     #[serde(flatten)]
     pub signature: Signature,
     pub quote_id: Option<QuoteId>,
+    pub full_app_data: Option<String>,
```

Comment:
While we expect the full app data to be a string, shouldn't we already deserialize instead of it being a string in the model here? We can use `serde` deserializers for that (meaning that you can't have an `OrderCreation::full_app_data` that is invalid).

Also, I'm not a fan of `full_app_data` as a name. I don't have any good suggestions beside `appData` (which would be a breaking change, but I think that this will have to be a breaking change anyway).

Reply:
Specifically an argument for not using a `String` here, the OpenAPI spec specifically says this is an `object`, but the current type doesn't encode that at all.

Reply:
Openai says that this is a string. I also added a FullAppData object to document the json schema. But it is a string.

We can't deserialize here because we need to store the exact signed bytes in the database.

Reply:
"full app data" is the name I used in the design issue. There is "app data hash" and "full app data". We shouldn't make a breaking change for the old name but I'm open for suggestions on changing the new name. Note that currently this isn't a breaking change because the field is optional and I would like to keep it a non breaking change when possible: Order created with all zero app data don't need to set full app data.

Reply:
> * I'm really not a fan of the name `full_app_data` at it seems imprecise (why is it full, how is "full" app data different than the "app_data")

This is explained in the api documentation. But the yes the field names are slightly misleading.

Reply:
I believe at some point Nick suggested to use the [JCS](https://www.rfc-editor.org/rfc/rfc8785) format. This would result in a deterministic JSON format. That could be stored as a `jsonb` in the DB and used to recreate the originally signed data.
This would probably mean that we have to introduce `serde_jcs` and a bunch of awkward constraints only to avoid this string in the API here. Also since we will presumably store the backend relevant app data information in a DB table being able to store the original signed data as `jsonb` is kind of pointless.

Reply:
I see what you mean now. Note that the orderbook already validates all these things and returns appropriate errors. Only validated json makes it into the DB. They just don't happen on the top level Deserialize derive. Where this would make a difference is if we had other rust code making use of model. That code would have an easier time because it can use the more type safe representation.

I'm hesitant to put too much domain logic in the model crate. We already have too much of that. I'd like it to be closer to the json level. It's similar to what Niksha advocates for with DTO objects. We can still a struct like you propose that is optimized for use in Orderbook (for example) but I'm not sure it should be in model.

Regardless of what representation we use in Rust code, this is a json string. Would it be ok to move this discussion to later, out of this PR? I'd like to have this logic in and I feel the intricacies of the Rust representation are not so important compared to that.

Reply:
> Would it be ok to move this discussion to later, out of this PR?

Reply:
OK - lets table this suggestion for now and keep `OrderCreation` more DTO like. I do think that if we start manipulating the `AppData` (i.e. once we start parsing it to verify pre- and post- interactions, special order types, etc.) then we should introduce a proper `AppData` abstraction.